src/loaders/FileLoader.js

```diff
@@ -116,38 +116,40 @@ class FileLoader extends Loader {
 
 					// periodically read data into the new stream tracking while download progress
 					const stream = new ReadableStream( {
-						start( controller ) {
+						start: ( controller ) => {
 
-							readData();
+							const readData = () => {
+
+								reader.read()
+									.then( ( { done, value } ) => {
 
-							function readData() {
+										if ( done ) {
 
-								reader.read().then( ( { done, value } ) => {
+											controller.close();
 
-									if ( done ) {
+										} else {
 
-										controller.close();
+											loaded += value.byteLength;
 
-									} else {
+											const event = new ProgressEvent( 'progress', { lengthComputable, loaded, total } );
+											for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
 
-										loaded += value.byteLength;
+												const callback = callbacks[ i ];
+												if ( callback.onProgress ) callback.onProgress( event );
 
```

Comment:
Can we revert these indentation changes? It makes it hard to follow what has meaningfully changed. Please take make a minimal change we can more easily follow the code adjustments.

Reply:
Updated